app/models/work.rb

```diff
@@ -183,13 +183,34 @@ def new_recipients_allow_gifts
     self.new_gifts.each do |gift|
       next if gift.pseud.blank?
       next if gift.pseud&.user&.preference&.allow_gifts?
-      next if self.challenge_assignments.map(&:requesting_pseud).include?(gift.pseud)
-      next if self.challenge_claims.reject { |c| c.request_prompt.anonymous? }.map(&:requesting_pseud).include?(gift.pseud)
+      next if challenge_bypass(gift)
 
       self.errors.add(:base, ts("%{byline} does not accept gifts.", byline: gift.pseud.byline))
```

Comment:
```suggestion
      self.errors.add(:base, ts("%{byline} does not accept gifts.", byline: gift.pseud.byline))
      throw :abort
```
So that it doesn't generate further errors.

Reply:
At the moment, I'm more partial to making it [an explicit exception](https://github.com/otwcode/otwarchive/pull/4721/files#diff-ebe838f9d71d26b6227d6d37c96a7e2d93fb68a80d55075c7f2374bfcb4688abR197-R198) to avoid accidentally hiding another error we might want to show.

But debatable.